homeassistant/components/sonos/sensor.py

```diff
@@ -107,3 +130,36 @@ def _poll_state(self) -> None:
 
     async def _async_fallback_poll(self) -> None:
         """Provide a stub for required ABC method."""
+
+
+class SonosFavoritesEntity(SensorEntity):
+    """Representation of a Sonos favorites info entity."""
+
+    _attr_entity_registry_enabled_default = False
+    _attr_icon = "mdi:star"
+    _attr_name = "Sonos Favorites"
+    _attr_native_unit_of_measurement = "items"
+    _attr_should_poll = False
+
+    def __init__(self, favorites: SonosFavorites) -> None:
+        """Initialize the favorites sensor."""
+        self.favorites = favorites
+        self._attr_unique_id = f"{favorites.household_id}-favorites"
+
+    async def async_added_to_hass(self) -> None:
+        """Handle common setup when added to hass."""
+        await self._async_update_state()
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{SONOS_FAVORITES_UPDATED}-{self.favorites.household_id}",
+                self._async_update_state,
+            )
+        )
+
+    async def _async_update_state(self) -> None:
+        self._attr_native_value = self.favorites.count
+        self._attr_extra_state_attributes = {
+            "items": {fav.item_id: fav.title for fav in self.favorites}
```

Comment:
Should these be excluded from the recorder?

Reply:
Yes. I guess in that case it needs to become a `sonos.favorites` entity

Reply:
Hmm they don't change often as it's limited to 72 entries (lol Sonos). I am not sure if this warrants enough reason to migrate from `sensor` domain to `sonos`.

Reply:
Actually it's 70 😆 and the state should only update if the user modifies the favorites from the Sonos app. I assumed that creating it as disabled by default would limit the impact.

Reply:
If we rename `items` to `favorites` and add the recorder platform to this integration to exclude that attribute from recording, it would be solved?

Alternatively, change this to be an `select` instead; That way it becomes a functional thing (as people can use it to select a favorite), option can be retrieved in templating AND the options lists of a `select` is excluded from being recorded.

Reply:
I started with a `select` entity, but changed my mind for a couple reasons:
1. It would require an entity per device instead of one per system and an update to favorites would write to X entities instead of one. However, it sounds like the options list isn't included in the recorder and this may be a non-issue.
2. What should a `select` entity state show if none of the available options are in use?

Reply:
I feel like a select would become a shitty version of the actual media player which offers favorites via the media browser. The goal of this entity is just to provide people this data for their templating.